crates/ruma-common/src/events/room/join_rules.rs

```diff
@@ -97,6 +103,11 @@ pub enum JoinRule {
     #[serde(rename = "restricted")]
     Restricted(Restricted),
 
+    /// Users can join the room if they are invited, or if they meet any of the conditions
+    /// described in a set of [`AllowRule`]s, or they can request an invite to the room.
+    #[serde(rename = "knock_restricted")]
```

Comment:
I wonder why we didn't use `rename_all` here.

Reply:
I don't know either. Do you want me to make this change in this PR?

Reply:
Whatever you prefer.